dcm4che-conf/dcm4che-conf-dicom/src/main/java/org/dcm4che3/conf/dicom/CommonDicomConfiguration.java

```diff
@@ -269,6 +270,41 @@ public Device findDeviceByUUID(String uuid) throws ConfigurationException {
         }
     }
 
+    @Override
+    public Device[] listDevices(DeviceType... primaryDeviceTypes) throws ConfigurationException {
+        if (primaryDeviceTypes == null) {
+            throw new IllegalArgumentException("Requested primaryDeviceTypes cannot be null");
```

Comment:
I don't think this should be here. I thought that this method could cope with a null primaryDeviceTypes (see interface declaration javadoc). The listDeviceNamesBy(primaryDeviceTypes) will return an empty list if primaryDeviceTypes is null (line 281).

Reply:
Now that you mentioned, I acknowledge this can be misleading :anguished:

_`primaryDeviceTypes Can be optionally included. If not, all the {@link Device}s will be retrieved`_
My intention here was to allow a value to either be included in the call or not, as below:
```
configuration.listDevices(); // We're passing an empty array here
configuration.listDevices(DeviceType.PRINT);
configuration.listDevices(DeviceType.PRINT, DeviceType.ARCHIVE);
```
The `IllegalArgumentException` would be thrown if the method is called by explicitly passing `null`:
```
configuration.listDevices(null);
```

When I pushed my latest changes I thought it would be risky allowing null arrays to be passed on, as we could end up having unintended results being retrieved. But this risk actually involves empty arrays the same way. :boom:

One thing I could do to make things safer and easier to use is to split this method according to the acceptable parameters, explicitly (no arguments, one DeviceType, multiple DeviceTypes). This way the caller would know exactly what to expect from each method:
```
Device[] listDevices(); // Will list all devices
Device[] listDevices(DeviceType primaryDeviceType); // List devices according to DeviceType - Null returns an empty Device array
Device[] listDevices(DeviceType[] primaryDeviceTypes); // List devices according to DeviceTypes - Null or empty array returns an empty Device array
```

Would you think this makes more sense?

Reply:
Yeah, I think overloading the method makes the more sense, though maybe the first method could be called listAllDevices(), just to be clear.

Reply:
Pushed c7ea96d to address these changes